pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -57,10 +57,12 @@ func (srv AlertmanagerSrv) loadSecureSettings(ctx context.Context, orgId int64,
 	currentReceiverMap := make(map[string]*apimodels.PostableGrafanaReceiver)
 	if query.Result != nil {
 		currentConfig, err := notifier.Load([]byte(query.Result.AlertmanagerConfiguration))
+		// If the current config is un-loadable, treat it as if it never existed. Providing a new, valid config should be able to "fix" this state.
 		if err != nil {
-			return fmt.Errorf("failed to load latest configuration: %w", err)
+			srv.log.Warn("Last known alertmanager configuration was invalid. Overwriting...")
+		} else {
+			currentReceiverMap = currentConfig.GetGrafanaReceiverMap()
```

Comment:
This is the actual fix. `loadSecureSettings` has a default behavior for if an org does not have an AM config, which is to treat things as a create rather than an update.

If the org _does_ have an AM config but it's broken, it would return an error. The result is that if an org had a broken AM config, any request to fix it would error and return a 500.

Instead, if the org has a broken/unusable config, treat the org the same as if it has no config. Let future requests fix the problem.